scheduler/replica_scheduler_test.go

```diff
@@ -0,0 +1,231 @@
+)
+
+func newReplicaScheduler(lhInformerFactory lhinformerfactory.SharedInformerFactory, kubeInformerFactory informers.SharedInformerFactory,
+	lhClient *lhfake.Clientset, kubeClient *fake.Clientset) *ReplicaScheduler {
+	fmt.Printf("testing NewReplicaScheduler\n")
+
+	volumeInformer := lhInformerFactory.Longhorn().V1alpha1().Volumes()
+	engineInformer := lhInformerFactory.Longhorn().V1alpha1().Engines()
+	replicaInformer := lhInformerFactory.Longhorn().V1alpha1().Replicas()
+	engineImageInformer := lhInformerFactory.Longhorn().V1alpha1().EngineImages()
+	nodeInformer := lhInformerFactory.Longhorn().V1alpha1().Nodes()
+
+	podInformer := kubeInformerFactory.Core().V1().Pods()
+	cronJobInformer := kubeInformerFactory.Batch().V1beta1().CronJobs()
+	daemonSetInformer := kubeInformerFactory.Apps().V1beta2().DaemonSets()
+
+	ds := datastore.NewDataStore(volumeInformer, engineInformer, replicaInformer, engineImageInformer, lhClient,
+		podInformer, cronJobInformer, daemonSetInformer, kubeClient, TestNamespace, nodeInformer)
+
+	return NewReplicaScheduler(ds)
+}
+
+func newDaemonPod(phase v1.PodPhase, name, namespace, nodeID, podIP string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "longhorn-manager",
+			},
+		},
+		Spec: v1.PodSpec{
+			NodeName: nodeID,
+		},
+		Status: v1.PodStatus{
+			Phase: phase,
+			PodIP: podIP,
+		},
+	}
+}
+
+func newNode(name, namespace string, allowScheduling bool, nodeState types.NodeState) *longhorn.Node {
+	return &longhorn.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: types.NodeSpec{
+			AllowScheduling: allowScheduling,
+			Disks: map[string]types.DiskSpec{
+				TestDefaultDataPath: {
+					Path: TestDefaultDataPath,
+				},
+			},
+		},
+		Status: types.NodeStatus{
+			State: nodeState,
+		},
+	}
+}
+
+func newVolume(name string, replicaCount int) *longhorn.Volume {
+	return &longhorn.Volume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Finalizers: []string{
+				longhorn.SchemeGroupVersion.Group,
+			},
+		},
+		Spec: types.VolumeSpec{
+			NumberOfReplicas:    replicaCount,
+			Size:                TestVolumeSize,
+			OwnerID:             TestOwnerID1,
+			StaleReplicaTimeout: TestVolumeStaleTimeout,
+			EngineImage:         TestEngineImage,
+		},
+	}
+}
+
+func newReplicaForVolume(v *longhorn.Volume) *longhorn.Replica {
+	return &longhorn.Replica{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: v.Name + "-r-" + util.RandomID(),
+			Labels: map[string]string{
+				"longhornvolume": v.Name,
+			},
+		},
+		Spec: types.ReplicaSpec{
+			InstanceSpec: types.InstanceSpec{
+				OwnerID:     v.Spec.OwnerID,
+				VolumeName:  v.Name,
+				VolumeSize:  v.Spec.Size,
+				EngineImage: TestEngineImage,
+				DesireState: types.InstanceStateStopped,
+			},
+		},
+	}
+}
+
+func TestReplicaScheduler_ScheduleReplica(t *testing.T) {
```

Comment:
Go style is CamelCase without underscore(`_`). Just do `TestReplicaScheduler` should be fine for now.